packages/x-license/src/verifyLicense/verifyLicense.ts

```diff
@@ -130,6 +130,10 @@ export function verifyLicense({
   licenseKey?: string;
   packageName: MuiCommercialPackageName;
 }): { status: LicenseStatus; meta?: any } {
+  if (location.host === 'mui.com') {
+    return { status: LICENSE_STATUS.Valid };
```

Comment:
This seems hacky. 🙈 

WDYT about reverting back to aliasing and looking into optimizing the license key management for docs before going back to using built packages. 🤔

Reply:
The [current setup](https://github.com/mui/mui-x/pull/17214#discussion_r2041850762) already seems hacky, it looks like we ended up with it by accident and it's hard to understand how/why it's setup as it is.

I've replaced this by a `MUI_DISABLE_LICENSE_CHECK` which explicitly builds a version of `x-license` without checks, just for the docs. I restored aliasing in the docs but only for that package.

Reply:
That env variable is resolved at build time, so the NPM packages won't contain it. It's essentially the same thing we do with `__RELEASE_INFO__` currently, but done explicitly instead of relying on an expired timestamp & license.

Reply:
Great job, nice optimization! 👏